tsdb/head.go

```diff
@@ -1451,17 +1456,17 @@ type memSeries struct {
 
 	ref           uint64
 	lset          labels.Labels
-	mmappedChunks []*mmappedChunk
-	mmMaxTime     int64 // Max time of any mmapped chunk, only used during WAL replay.
-	headChunk     *memChunk
+	mmappedChunks []*mmappedChunk // Immutable chunks on disk that have not yet gone into a block, in order of ascending time stamps.
+	mmMaxTime     int64           // Max time of any mmapped chunk, only used during WAL replay.
+	headChunk     *memChunk       // Most recent chunk in memory that's still being built.
 	chunkRange    int64
-	firstChunkID  int
+	firstChunkID  int // ChunkID corresponding to mmappedChunks[0].ref
 
-	nextAt        int64 // Timestamp at which to cut the next chunk.
-	sampleBuf     [4]sample
-	pendingCommit bool // Whether there are samples waiting to be committed to this series.
+	nextAt        int64     // Timestamp at which to cut the next chunk.
+	sampleBuf     [4]sample // We keep the last 4 samples here (in addition to appending them) so we don't need coordination between appender and querier. Probably because even the most compact encoding of a sample takes 2 bits, so the last byte is not contended.
+	pendingCommit bool      // Whether there are samples waiting to be committed to this series.
 
-	app chunkenc.Appender // Current appender for the chunk.
+	app chunkenc.Appender // Current appender for the head chunk. Set when a new head chunk is cut. TODO when exactly is it nil or not nil? is it valid to be nil?
```

Comment:
It is nil only if headChunk is nil. And that is possible if there was an appender that created a new series, but rolled back the commit (the first sample would create a headChunk, hence appender, but rollback skipped it while the Append() call would create a series).
```suggestion
	app chunkenc.Appender // Current appender for the head chunk. Set when a new head chunk is cut.
```